### PR TITLE
libs/libc/aio: fix aio_write compatible bug

### PR DESCRIPTION
**Pull request by extinguish**
## Summary
1. make the aio_write implementation can pass the lpt/open_posix_testsuite/aio_write testcases
2. the modification are referred to https://pubs.opengroup.org/onlinepubs/9699919799/functions/aio_write.html

### CODE REVIEW

**Review by xiaoxiang781216 — APPROVED** [review]

**Comment by xiaoxiang781216** [comment]
Let's ignore macOS CI broken due to https://github.com/apache/nuttx-apps/issues/1812.

File: fs/aio/aio_write.c [hunk 1]
```diff
@@ -249,6 +249,40 @@ int aio_write(FAR struct aiocb *aiocbp)
...
+      aiocbp->aio_result = -EBADF;
+      return OK;
+    }
+
+  /* the aio_fildes that transferred in may be opened with O_RDONLY, for this
+   * case, we need to return OK directly, and using the aio_error to collect
+   * the EBADF error code
+   */
+
+  int flags = fcntl(aiocbp->aio_fildes, F_GETFL);
```
Comment (pkarashchenko) [inline fs/aio/aio_write.c:279]: C89 incompatible

File: fs/aio/aio_write.c [hunk 2]
```diff
@@ -249,6 +249,40 @@ int aio_write(FAR struct aiocb *aiocbp)
...
+  if (!(flags & O_WRONLY))
```
Comment (pkarashchenko) [inline fs/aio/aio_write.c:280]: Optional
```suggestion
  if ((flags & O_WRONLY) == 0)
```

File: fs/aio/aio_write.c (hunk 1 above)
Comment (extinguish) [reply fs/aio/aio_write.c:279]: But I use gcc test.c -std=c89 -o test -lrt

File: fs/aio/aio_write.c (hunk 2 above)
Comment (extinguish) [reply fs/aio/aio_write.c:280]: fixed in https://github.com/apache/nuttx/pull/9655

**Pull request merged**